pkg/server/api/entry.go

```diff
@@ -54,16 +54,59 @@ func ProtoToRegistrationEntry(e *types.Entry) (*common.RegistrationEntry, error)
 		return nil, errors.New("missing entry")
 	}
 
-	parentID, err := IDFromProto(e.ParentId)
+	_, err := IDFromProto(e.ParentId)
```

Comment:
Hmm, this seems a little unideal as it ends up processing the parentId and spiffeID twice (once here and once down to protoToRegistrationEntry.

I guess what I expected to see here was:
1) `ProtoToRegistrationEntryWithMask` which takes a `*types.Entry` and `*types.EntryMask`. It converts a `nil` mask to an all-true mask. It then creates a `*common.RegistrationEntry`, and for each field enabled by the mask, converts and sets it on the `common` entry, failing if any of those fields are required by empty in the `*types.Entry`.
2) `ProtoToRegistrationEntry` is a wrapper that calls `ProtoToRegistrationEntryWithMask` with a `nil` mask, which means that all fields will be converted and evaluated against required field requirements.